lib/core/configurator.c

```diff
@@ -1259,7 +1259,8 @@ int h2o_configurator__do_parse_mapping(h2o_configurator_command_t *cmd, yoml_t *
             return -1;
         }
         if ((keys[j].type_mask & (1u << element->value->type)) == 0) {
-            char permitted_types[32] = "";
+            char permitted_types[41] = "";
+            assert(sizeof(permitted_types) >= strlen(" or a scalar or a sequence or a mapping") + 1);
```

Comment:
How about doing:
```c
snprintf(permitted_types, sizeof(permitted_types), "%s%s%s",
         (key[j].type_mask & (1u << YOML_TYPE_SCALAR) != 0 ? "..." : "",
         (key[j].type_mask & (1u << YOML_TYPE_ARRAY) != 0 ? "..." : "",
         (key[j].type_mask & (1u << YOML_TYPE_MAPPING) != 0 ? "..." : "");
```

That would remove the need to check the size.

Reply:
Yeah this is much cleaner. Will do so.

Reply:
Done in 7ab87e71dcd4569a28ec18b68a60e194e8297ff0.